src/app/components/add-task/add-task.component.ts

```diff
@@ -19,25 +22,31 @@ export class AddTaskComponent implements OnInit {
     "Task"
   ]
   @ViewChild('addformtemplate') popupContent?: TemplateRef<any>;
-  formGroup: FormGroup | null = null;
+  formGroup!: FormGroup;
+
   constructor(private formBuilder: FormBuilder,
       private dialogRef: MatDialog,
-      private taskService: TaskService) { }
+      private taskService: TaskService,
+      private taskFormValidationService: TaskFormValidationService) { }
 
   ngOnInit(): void {
     this.formGroup = this.formBuilder.group({
-      title: [""],
+      title: ["", {validators: [Validators.required,
+        this.taskFormValidationService.uniqueTitle(),
+      ]}],
       description: [""],
-      type: [""]
+      type: ["", {validators: [Validators.required]}]
```

Comment:
curly braces not necessary here, since in the array 1st item is value of the form control and 2nd is either just 1 validator or you can just add an array of validators so it'd look something like this:
[your_initial_value, [Validators.email, Validators.required]]